src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,131 @@
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<>(lastListElement, value, null);
+        if (size == 0) {
+            firstListElement = node;
+        } else {
+            lastListElement.next = node;
+        }
+        lastListElement = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkNodeInList(index);
+        Node<T> oldNode = getNodeFromList(index);
+        Node<T> newNode = new Node<>(oldNode.prev, value, oldNode);
+        if (index > 0) {
+            newNode.prev.next = newNode;
+        } else {
+            firstListElement = newNode;
+        }
+        oldNode.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkNodeInList(index);
+        Node<T> node = getNodeFromList(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkNodeInList(index);
+        Node<T> node = getNodeFromList(index);
+        T oldNodeValue = node.item;
+        node.item = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkNodeInList(index);
+        Node<T> node = getNodeFromList(index);
+        if (index > 0) {
+            node.prev.next = node.next;
+        } else {
+            firstListElement = node.next;
+        }
+        if (node.next != null) {
+            node.next.prev = node.prev;
+        } else {
+            lastListElement = node.prev;
+        }
+        T removedItem = node.item;
+        size--;
+        return removedItem;
     }
 
     @Override
     public boolean remove(T t) {
+        Node<T> node = firstListElement;
+        for (int i = 0; i < size; i++) {
+            if (node.item == t || node.item.equals(t)) {
+                remove(i);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private boolean checkNodeInList(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        return true;
+    }
+
+    private Node<T> getNodeFromList(int index) {
+        Node<T> node = firstListElement;
+        for (int i = 0; i < index; i++) {
```

Comment:
let's iterate our linkedlist depending on location of our node: 
 - if node is closer to the start, we will go from first to middle, if it' closer to the end of list we will go from last to middle